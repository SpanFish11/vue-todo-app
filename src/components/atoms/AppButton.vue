<script setup>
const props = defineProps({
  label: { type: String, required: false, default: '' },
  customClass: { type: Array, required: false },
})

const emit = defineEmits(['onClickButton'])
</script>

<template>
  <button :class="[customClass]" @click="emit('onClickButton')">
    {{ props.label }}
  </button>
</template>
