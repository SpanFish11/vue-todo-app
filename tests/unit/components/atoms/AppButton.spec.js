import { beforeEach, describe, expect, it } from '@jest/globals'
import { shallowMount } from '@vue/test-utils'
import AppButton from '@/components/atoms/AppButton.vue'

describe('Given the AppButton component', () => {
  let wrapper

  beforeEach(() => {
    wrapper = shallowMount(AppButton, {})
  })

  it.each([
    [{ label: 'label', customClass: ['custom-class-1', 'custom-class-2'] }],
    [{ customClass: ['custom-class-1', 'custom-class-2'] }],
    [{ label: 'label' }],
    [{}],
  ])('should render', (props) => {
    wrapper.setProps(props)

    expect(wrapper.element).toMatchSnapshot()
  })

  it('should emits a click event when button is clicked', async () => {
    await wrapper.find('button').trigger('click')
    const onClickButtonEvent = wrapper.emitted('onClickButton')

    expect(wrapper.emitted()).toHaveProperty('onClickButton')
    expect(onClickButtonEvent).toHaveLength(1)
  })
})
