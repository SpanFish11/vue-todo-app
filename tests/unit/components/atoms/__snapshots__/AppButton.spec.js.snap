// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Given the AppButton component should render 1`] = `
<button
  class=""
/>
`;

exports[`Given the AppButton component should render 2`] = `
<button
  class=""
/>
`;

exports[`Given the AppButton component should render 3`] = `
<button
  class=""
/>
`;

exports[`Given the AppButton component should render 4`] = `
<button
  class=""
/>
`;
